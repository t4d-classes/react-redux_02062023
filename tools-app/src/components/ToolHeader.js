import PropTypes from 'prop-types';


export function ToolHeader(props) {

  return (
    <header>
      <h1>{props.appName}</h1>
      <h2>{props.toolName}</h2>
    </header>
  );

}

ToolHeader.defaultProps = {
  appName: 'Tools App',
};

ToolHeader.propTypes = {
  appName: PropTypes.string.isRequired,
  toolName: PropTypes.string,
};

// export function ToolHeader({ appName, toolName }) {

//   return (
//     <header>
//       <h1>{appName}</h1>
//       <h2>{toolName}</h2>
//     </header>
//   );

// }