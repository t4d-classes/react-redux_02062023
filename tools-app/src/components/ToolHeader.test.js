import React from "react";

import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import { ToolHeader } from "./ToolHeader";

test("snapshot ToolHeader component", () => {
  expect(
    renderer.create(
      <ToolHeader appName="The Tools" toolName="Test Tool" />
    ).toJSON()
  ).toMatchSnapshot();
});

describe("ToolHeader component", () => {
  test("renders ToolHeader component", () => {
    render(
      <ToolHeader appName="The Tools" toolName="Test Tool" />
    );

    expect(screen.getByText("The Tools")).toBeInTheDocument();
    expect(screen.getByText("Test Tool")).toBeInTheDocument();

    return screen.findAllByRole('heading').then((elements) => {
      expect(elements.length).toEqual(2);
      expect(elements[0].textContent).toEqual("The Tools");
      expect(elements[1].textContent).toEqual("Test Tool");
    });
  });
});