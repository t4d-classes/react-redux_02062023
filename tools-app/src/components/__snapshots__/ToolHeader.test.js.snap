// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ToolHeader Component 1`] = `
<header>
  <h1>
    The Tools
  </h1>
  <h2>
    Test Tool
  </h2>
</header>
`;
